File: Python3/프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    
    stack = []
    
    for i in range(len(arr)):
        
        if not stack:
            stack.append(arr[i])
        else:
            if stack[-1] == arr[i]:
                continue
            else:
                stack.append(arr[i])
    
    return stack
            

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int[] arr) {
        List<Integer> list = new ArrayList<>();
        
        Stack<Integer> stack = new Stack<>();
        for(int i=0; i<arr.length; i++) {
            if(stack.isEmpty()) {
                list.add(arr[i]);
                stack.push(arr[i]);
            } else {
                if(stack.peek() != arr[i]) {
                    list.add(arr[i]);
                    stack.push(arr[i]);
                }
            }
        }
        int[] res = new int[list.size()];
        for(int i=0; i<res.length; i++) {
            res[i] = list.get(i);
        }
        
        return res;
    }
}